lib/connection_pool/connection_wrapper.rb

```diff
@@ -0,0 +1,20 @@
+require_relative 'monotonic_time'
+
+class ConnectionPool::ConnectionWrapper
+  attr_reader :raw_conn
+
+  def initialize(conn, connection_manager)
```

Comment:
`connection_pool` as a single argument should be enough.
This class should ask pool for raw_conn on initialize.
Also it may incapsulate logic for managing connection. I see methods like `connected?` `reconnect_if_max_age` (or something like this).